integration-test/src/main/java/com/sequenceiq/it/cloudbreak/listener/ReportListener.java

```diff
@@ -81,6 +115,108 @@ private void logMeasurements(ITestResult tr) {
         }
         tr.setAttribute(tr.getName() + MEASUREMENTS, measuredTestContext.getMeasure());
     }
+
+    private void generateFreeIpaLogsUrl(ITestResult tr, Map<String, CloudbreakTestDto> resourceNames,
+            Map<String, CloudbreakTestDto> resourceCrns, String baseLocation, CloudProviderProxy cloudProvider) {
+        StorageUrl storageUrl = new ClusterLogsStorageUrl();
+
+        try {
+            String resourceName = getResourceName(FreeIpaTestDto.class, resourceNames);
+            String resourceCrn = getResourceCrn(FreeIpaTestDto.class, resourceCrns);
+            if (StringUtils.isEmpty(resourceCrn)) {
+                LOGGER.info("FreeIPA resource is not present in Cluster Logs");
+            } else {
+                String freeIpaCloudStorageUrl = storageUrl.getFreeIpaStorageUrl(resourceName, resourceCrn, baseLocation, cloudProvider);
+                tr.getTestContext().setAttribute(tr.getName() + FI_STORAGE_URL, freeIpaCloudStorageUrl);
+                LOGGER.info("FreeIpa cloud storage: {}:{}", tr.getName() + FI_STORAGE_URL, freeIpaCloudStorageUrl);
+            }
+        } catch (Exception e) {
+            LOGGER.info("FreeIPA resource is not present in Test Context.", e);
+        }
+    }
```

Comment:
these methods are quite similar, is it possible to create one common method with for them?